src/http/router.ts

```diff
@@ -718,42 +718,44 @@ export const getRouter = (client: CLClient, options: ICLClientOptions): express.
-                    const scoreInput: CLM.ScoreInput = {
-                        filledEntities,
-                        context: {},
-                        maskedActions: []
-                    }
-                    teachWithHistory.scoreInput = scoreInput
-                    teachWithHistory.scoreResponse = await client.TeachScore(
-                        appId,
-                        teachWithHistory.teach.teachId,
-                        teachWithHistory.scoreInput
-                    )
-                }
-                else if (userInput) {
-                    // Add new input to history
-                    let userActivity = CLM.ModelUtils.InputToActivity(userInput.text, userName, userId, trainDialog.rounds.length)
-                    teachWithHistory.history.push(userActivity)
-
-                    // Extract responses
-                    teachWithHistory.extractResponse = await client.TeachExtract(appId, teachWithHistory.teach.teachId, userInput, null)
-                    teachWithHistory.dialogMode = CLM.DialogMode.Extractor
+            // Start new teach session from the old train dialog
+            const createTeachParams = CLM.ModelUtils.ToCreateTeachParams(cleanTrainDialog)
+            teachWithHistory.teach = await clRunner.StartSessionAsync(clMemory, null, appId, SessionStartFlags.IN_TEACH | SessionStartFlags.IS_EDIT_CONTINUE, createTeachParams) as CLM.Teach
+
+            // If last action wasn't terminal then score
+            if (teachWithHistory.dialogMode === CLM.DialogMode.Scorer) {
+
+                // Get entities from my memory
+                const filledEntities = await clMemory.BotMemory.FilledEntitiesAsync()
+                const scoreInput: CLM.ScoreInput = {
+                    filledEntities,
+                    context: {},
+                    maskedActions: []
                 }
-                res.send(teachWithHistory)
-            } catch (error) {
-                HandleError(res, error)
+                teachWithHistory.scoreInput = scoreInput
+                teachWithHistory.scoreResponse = await client.TeachScore(
+                    appId,
+                    teachWithHistory.teach.teachId,
+                    teachWithHistory.scoreInput
+                )
+            }
+            else if (userInput) {
+                // Add new input to history
+                let userActivity = CLM.ModelUtils.InputToActivity(userInput.text, userName, userId, trainDialog.rounds.length)
+                teachWithHistory.history.push(userActivity)
+
+                // Extract responses
+                teachWithHistory.extractResponse = await client.TeachExtract(appId, teachWithHistory.teach.teachId, userInput, null)
+                teachWithHistory.dialogMode = CLM.DialogMode.Extractor
             }
+            res.send(teachWithHistory)
         } catch (error) {
+            if (error.statusCode === HttpStatus.CONFLICT) {
```

Comment:
This diff looks a lot worse than the change.  This addition of `if` statement to catch conflicts in the `catch` block was the only change that matters.  The other differs were removing an unecessary `try` / `catch`.  Both the outer and inner blocks had the same `catch` definition so the outer already suffices.